h/accounts/views.py

```diff
@@ -115,27 +116,34 @@ def wrapper(context, request):
 @view_auth_defaults
 @view_config(attr='login', route_name='login')
 @view_config(attr='logout', route_name='logout')
-class AuthController(horus.views.AuthController):
+class AuthController(object):
+    def __init__(self, request):
+        self.request = request
+        self.schema = schemas.LoginSchema().bind(request=self.request)
+        self.form = deform.Form(self.schema)
+
+        self.login_redirect = self.request.route_url('stream')
+        self.logout_redirect = self.request.route_url('index')
+
     def login(self):
         if self.request.authenticated_userid is not None:
-            return httpexceptions.HTTPFound(location=self.login_redirect_view)
+            return httpexceptions.HTTPFound(location=self.login_redirect)
 
-        try:
-            user = self.form.validate(self.request.POST.items())['user']
-        except deform.ValidationFailure as e:
-            return {
-                'status': 'failure',
-                'errors': e.error.children,
-                'reason': e.error.msg,
-            }
+        err, appstruct = validate_form(self.form, self.request.POST.items())
+        if err is not None:
+            return err
```

Comment:
I've seen this pattern a few times. Just a thought: would it be nice if controller methods could return an error to the browser by raising an exception, rather than by returning an error object? Because then they could just blindly call `appstruct = validate_form()` which could raise the exception for them if validation fails.